code/game/objects/random/misc.dm

```diff
@@ -932,3 +932,39 @@
 				prob(3);/obj/item/clothing/accessory/poncho/thermal/purple,
 				prob(3);/obj/item/clothing/accessory/poncho/thermal/blue)
 
+/obj/random/mug
+	name = "Random Mug"
+	desc = "This is a random coffee mug."
+	icon = 'icons/obj/drinks_mugs.dmi'
+	icon_state = "coffeecup_spawner"
+
+/obj/random/mug/item_to_spawn()
```

Comment:
May be preferable here to simply `return pick(typesof(/obj/item/weapon/reagent_containers/food/drinks/glass2/coffeemug))` since it will mean not having to maintain the list later.